# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/lld/hotfix-no-macho.patch
# Copyright (C) 2022 - 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

lld-17.0.1.src/MachO/Target.h:23:10: fatal error: mach-o/compact_unwind_encoding.h: No such file or directory
   23 | #include "mach-o/compact_unwind_encoding.h"

--- lld-15.0.1.src/CMakeLists.txt.vanilla	2022-09-26 15:16:47.697000000 +0200
+++ lld-15.0.1.src/CMakeLists.txt	2022-09-26 15:16:52.630000000 +0200
@@ -256,7 +256,6 @@
 add_subdirectory(docs)
 add_subdirectory(COFF)
 add_subdirectory(ELF)
-add_subdirectory(MachO)
 add_subdirectory(MinGW)
 add_subdirectory(wasm)
 
--- lld-17.0.1.src/include/lld/Common/Driver.h.vanilla	2023-09-19 15:08:59.613505175 +0200
+++ lld-17.0.1.src/include/lld/Common/Driver.h	2023-09-19 15:09:51.209501783 +0200
@@ -61,7 +61,7 @@
 #define LLD_ALL_DRIVERS                                                        \
   {                                                                            \
     {lld::WinLink, &lld::coff::link}, {lld::Gnu, &lld::elf::link},             \
-        {lld::MinGW, &lld::mingw::link}, {lld::Darwin, &lld::macho::link}, {   \
+        {lld::MinGW, &lld::mingw::link}, {   \
       lld::Wasm, &lld::wasm::link                                              \
     }                                                                          \
   }
--- lld-17.0.1.src/tools/lld/lld.cpp.vanilla	2023-09-19 15:06:08.856516402 +0200
+++ lld-17.0.1.src/tools/lld/lld.cpp	2023-09-19 15:07:02.515512874 +0200
@@ -71,7 +71,6 @@
 LLD_HAS_DRIVER(coff)
 LLD_HAS_DRIVER(elf)
 LLD_HAS_DRIVER(mingw)
-LLD_HAS_DRIVER(macho)
 LLD_HAS_DRIVER(wasm)
 
 int lld_main(int argc, char **argv, const llvm::ToolContext &) {
--- lld-15.0.1.src/tools/lld/CMakeLists.txt.vanilla	2022-09-26 15:15:30.697000000 +0200
+++ lld-15.0.1.src/tools/lld/CMakeLists.txt	2022-09-26 15:15:34.541000000 +0200
@@ -14,7 +14,6 @@
   lldCommon
   lldCOFF
   lldELF
-  lldMachO
   lldMinGW
   lldWasm
   )
